notification-manager/src/main/java/com/qpay/notificationmanager/service/impl/EmailNotificationService.java

```diff
@@ -3,14 +3,25 @@
 import com.qpay.libs.models.PaymentNotification;
 import com.qpay.notificationmanager.service.NotificationService;
 import lombok.RequiredArgsConstructor;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class EmailNotificationService implements NotificationService {
 
-    public PaymentNotification sendMessage(final PaymentNotification paymentNotification) {
-        return paymentNotification;
+    private final JavaMailSender javaMailSender;
+
+    private final SimpleMailMessage paymentMessageTemplate;
+
+    public void sendMessage(final PaymentNotification paymentNotification) {
+        final SimpleMailMessage message = new SimpleMailMessage(paymentMessageTemplate);
+        final String text = String.format(paymentMessageTemplate.getText(), paymentNotification.amount());
```

Comment:
for me it looks complicated and hard to read. you create a template in a factory. Then recreate it and so on. Also your email from and all other text is hardcoded.

I think it's better to create a properties for each template.
```
email-templates:
  payment:
    from: "asd@gmail.com"
    subject: "sad"
    text: asdsad %s asdsad
    ...
  ```

 Then you can create a separate class `MailMessageGenerator` where you inject EmailTemplatesProperties class. Each method in `MailMessageGenerator` is responsible for dedicated template. 
 
 after that your current class can look like it:
 ```public void sendMessage(final PaymentNotification paymentNotification) {
        final var message = mailMessageGenerator.getPaymentMessage(paymentNotification);
        javaMailSender.send(message);
    }
   ```
   
   As a result, you data is flexible. You can change text anytime without changing the code.
   And it's easy to test each class, since you split the logic for creating mail, sending and so on.